warehouse/accounts/services.py

```diff
@@ -226,6 +250,30 @@ def is_disabled(self, user_id):
         else:
             return (True, user.disabled_for)
 
+    def has_two_factor(self, user_id):
+        """
+        Returns True if the user has any form of two factor
+        authentication.
+        """
+        two_factor = self.get_two_factor(user_id)
```

Comment:
Likewise to the above, ``two_factor`` can be ``None`` here, which will generate a less than ideal exception below.